### PR TITLE
Merge upstream 3.4.2

### PR DESCRIPTION
**Pull request by manicmaniac**
## Overview

- Merged upstream/3.4.2 at https://github.com/quipper/MessageKit/pull/24/commits/a92f80b4e2dba3b3267d6b58c452ade5a867862e.
- Resolved some conflicts at https://github.com/quipper/MessageKit/pull/24/commits/3a2ac0fd90e47b079356fc453fefc9b7cadbe34b
- Added missing `import UIKit` at https://github.com/quipper/MessageKit/pull/24/commits/304483d83f6e24cda52c11ef7a2190000186cd5b

3a2ac0fd90e47b079356fc453fefc9b7cadbe34b and 304483d83f6e24cda52c11ef7a2190000186cd5b need to be reviewed.

## Screenshot

image

### CODE REVIEW

File: Example/Podfile.lock [hunk 1]
```diff
@@ -47,10 +47,6 @@ SPEC CHECKSUMS:
   PINRemoteImage: 3b7cedb118c2d357f87e9eabc7c81ba0202cb236
   SwiftLint: dfd554ff0dff17288ee574814ccdd5cea85d76f7
 
-<<<<<<< HEAD
-PODFILE CHECKSUM: 05b7c748c0f30a6c617d2a85ead3fedcf0931a30
-=======
-PODFILE CHECKSUM: 24c39a08ead4a8771aa94310cfc35fe3be6b3105
->>>>>>> 3.4.2
+PODFILE CHECKSUM: 31c6eb1d734807f78336ea60201a116e2410b2dd
```
Comment (manicmaniac) [inline Example/Podfile.lock:50]: This checksum was recalculated by `bundle exec pod install`

File: Sources/Extensions/MessageKind+textMessageKind.swift [hunk 2]
```diff
@@ -29,7 +29,7 @@ internal extension MessageKind {
         switch self {
         case .linkPreview(let linkItem):
             return linkItem.textKind
-        case .text, .emoji, .attributedText:
+        case .announcement, .text, .emoji, .attributedText:
```
Comment (manicmaniac) [inline Sources/Extensions/MessageKind+textMessageKind.swift:32]: `.announcement` was added by us.

File: Sources/Layout/AnnounceMessageSizeCalculator.swift [hunk 3]
```diff
@@ -7,6 +7,7 @@
 //
 
 import Foundation
+import UIKit
```
Comment (manicmaniac) [inline Sources/Layout/AnnounceMessageSizeCalculator.swift:10]: From some version of Swift Package Manager, packages are not be compiled without declaring the correct imported modules.

In this file `UIFont`, `UIEdgeInsets` and other `UI`-prefixed classes are used.

**Review by elliekwon — APPROVED** [review]
https://github.com/quipper/MessageKit/commit/3a2ac0fd90e47b079356fc453fefc9b7cadbe34b と https://github.com/quipper/MessageKit/commit/304483d83f6e24cda52c11ef7a2190000186cd5b を確認致しました🙏

**Review by m-sugawara — APPROVED** [review]
LGTM 👍

**Review by k-kohey — APPROVED** [review]
LGTM

**Pull request merged**